site-scrapers/TrinityEMS/index.js

```diff
@@ -0,0 +1,253 @@
+            }
+        } catch (error) {
+            console.log(
+                `${site.name} :: error trying to get day numbers: ${error}`
+            );
+        }
+    }
+
+    return monthlyAvailability;
+}
+
+/**
+ * Looks for "no-dates-available" class, or ".activeday" elements.
+ * If not found, there are active days, and the page should be
+ * parsed for which days have availability.
+ *
+ * @returns Array of active day elements. Empty array is returned if there's no availability.
+ */
+async function getActiveDays(page) {
+    const noDatesAvailable = await page.$(".no-dates-available");
+    // No appointments available if defined
+    if (noDatesAvailable != undefined) {
+        return [];
+    } else {
+        let days = [];
+        try {
+            days = await page.$$(".activeday");
+        } catch (error) {
+            console.log(`error trying to get day numbers: ${error}`);
+        }
+        return days;
+    }
+}
+
+/**
+ * Fetches the small snippet of HTML which would appear in a popup when an active day is clicked by a human.
+ *
+ * As yet, it's not know how the slot count will be presented. See parseHTMLforSlotCount().
+ * @param {*} page
+ * @param {String} dateStr -- yyyy-mm-dd, as in '2021-04-26'
+ * @returns response text of available times XHR query
+ */
+async function getSlotsForDate(page, dateStr) {
+    const url = [
+        "https://app.acuityscheduling.com/schedule.php?action=availableTimes",
+        "showSelect=0",
+        "fulldate=1",
+        "owner=21713854",
+        "type=19620839",
+        "calendar=5109380",
+        `date=${dateStr}`,
+        "ignoreAppointment=",
+    ].join("&");
+
+    return await page.evaluate(async (url) => {
+        const response = await fetch(url);
+        const text = await response.text();
+        return parseHTMLforSlotCount(text);
+    }, url);
+}
+
+/**
+ * Advances the page to the next month by clicking on the "next" button.
+ *
+ * @param {*} page
+ */
+async function advanceMonth(page) {
+    const nextMonthLinkClass = ".calendar-next";
+
+    try {
+        await Promise.all([
+            await page.click(nextMonthLinkClass),
+            // If the timeout option is not set, the timeout (30000 ms) will expire, and
+            // no results will be returned. Setting it to something shorter yields results.
+            // 5000 seems to be too much because the pages return nearly instantaneously.
+            // But will leave it at that just in case of ... what?
+            await page.waitForSelector(nextMonthLinkClass, {
+                timeout: 5000,
+            }),
+        ]);
+    } catch (error) {
+        console.log(`${site.name} :: trouble advancing button: ${error}`);
+    }
+}
+
+/**
+ * Accumulates a month's worth of availability into the results object.
+ *
+ * @param {availability Object} dailySlotsForMonth -- a month's collection of availability by date
+ */
+function accumulateAvailabilityForMonth(results, dailySlotsForMonth) {
```

Comment:
This function could be refactored to be more functional with Javascript's array .`reduce` function, which I think reduces the chances of bugs. 
This article provides examples I find helpful: https://medium.com/poka-techblog/simplify-your-javascript-use-map-reduce-and-filter-bd02c593cc2d
I really prefer the more functional approach but I won't block approval on it

Reply:
It seems that Object.assign() is pretty succinct compared to a reduce statement (probably faster too). Would you object to the almost equally succinct ```results.availability = {...results.availability, ...dailySlotsForMonth}```? Or ```lodash.assign(results.availability, dailySlotsForMonth)```?

Reply:
Apologies for the delayed response. 

I was actually suggesting that you refactor the whole `accumulateAvailabilityForMonth` function to be return something (not void) so that you reduce the side-effects, which can make it harder to debug issues. But this is a style choice that I won't force you to follow (I would approve regardless of whether or not you refactor). 

As far as using `Object.assign()` vs the spread operator, I would suggest using the spread operator for the reasons laid out here: https://thecodebarbarian.com/object-assign-vs-object-spread.html 
and here https://medium.com/@corinnemariekelly/object-assign-vs-spread-operator-577c889dbadc
In particular, using the spread operator lets you avoid mutating the original object (which can again make things easier to debug).

To make it even more robust and clear what you're doing here, I would also suggest adding unit tests, but this can come in follow-up PR if you would like! You can use `EastBostonNHC` as an example of code and tests.